src/test/kotlin/com/zepben/evolve/cim/iec61970/base/core/IdentifiedObjectTest.kt

```diff
@@ -45,48 +45,133 @@ internal class IdentifiedObjectTest {
+    internal fun `getName obtain expected Name object`() {
+        val (identifiedObject: IdentifiedObject, nameType: NameType) = createMultipleBaseNames()
+
+        val name1 = identifiedObject.getName("type", "1")!!
+        val name2 = identifiedObject.getName("type", "2")!!
+        val name3 = identifiedObject.getName("type", "3")!!
 
+        assertThat(name2.name, equalTo("2"))
+        assertThat(name2.type.name, equalTo("type"))
+
+        // Make sure item obtained are different
         assertThat(name1, not(equalTo(name2)))
         assertThat(name1, not(equalTo(name3)))
         assertThat(name2, not(equalTo(name3)))
+    }
 
-        assertThat(identifiedObject.numNames(), equalTo(0))
+    @Test
+    internal fun `getName(String, String) and getName(NameType, String) grabs the same object`() {
+        val (identifiedObject: IdentifiedObject, nameType: NameType) = createMultipleBaseNames()
 
-        identifiedObject.addName(name1)
-        identifiedObject.addName(name2)
-        identifiedObject.addName(name3)
-        assertThat(identifiedObject.numNames(), equalTo(3))
-        identifiedObject.addName(duplicate1)
+        val name2 = identifiedObject.getName("type", "2")!!
+        val dupeName2 = identifiedObject.getName(nameType, "2")
+        assertThat(name2, sameInstance(dupeName2))
+    }
 
-        assertThat(identifiedObject.numNames(), equalTo(3))
+    @Test
+    internal fun `nameType contains the names added to the identified object`() {
+        val (identifiedObject: IdentifiedObject, nameType: NameType) = createMultipleBaseNames()
+
+        assertThat(nameType.hasName("1"), equalTo(true))
+        assertThat(nameType.hasName("2"), equalTo(true))
+        assertThat(nameType.hasName("3"), equalTo(true))
+    }
+
+    @Test
+    internal fun `removeName removes name from the identified object and the nameType`() {
+        val (identifiedObject: IdentifiedObject, nameType: NameType) = createMultipleBaseNames()
+        val name1 = identifiedObject.getName("type", "1")!!
+        val name2 = identifiedObject.getName("type", "2")!!
+        val name3 = identifiedObject.getName("type", "3")!!
 
         assertThat(identifiedObject.removeName(name1), equalTo(true))
         assertThat(identifiedObject.removeName(name1), equalTo(false))
         assertThat(identifiedObject.removeName(null), equalTo(false))
         assertThat(identifiedObject.numNames(), equalTo(2))
-
-        assertThat(identifiedObject.getName("type", name2.name), equalTo(name2))
+        assertThat(nameType.hasName("1"), equalTo(false))
 
         assertThat(identifiedObject.names, containsInAnyOrder(name2, name3))
+    }
+
+    @Test
+    internal fun `clearNames removes all names from the identified object and the nameType`() {
+        val (identifiedObject: IdentifiedObject, nameType: NameType) = createMultipleBaseNames()
+
+        assertThat(identifiedObject.numNames(), equalTo(3))
+        assertThat(nameType.hasName("1"), equalTo(true))
```

Comment:
chore: Can you please update the tests in the form of `assertThat(x, equalTo(true/false))` to `assertThat("description", x)`. You have just copied the pattern that was already in this file, so can you also please update the other calls as I am trying to migrate all of these types of tests over time.

e.g.:
`assertThat(nameType.hasName("1"), equalTo(true))` becomes `assertThat("expected to have name 1", nameType.hasName("1"))`
`assertThat(nameType.hasName("1"), equalTo(false))` becomes `assertThat("should not have had name 1", !nameType.hasName("1"))`

Reply:
Done. And I've gone through the nametype test and did the same because I had to update the test to reflect the two way relation for identifiedObject and nameType when removing/adding names.